src/main/scala/com/datastax/sparkstress/StreamingTask.scala

```diff
@@ -25,7 +25,21 @@ abstract class StreamingTask[rowType](
 
   val ssc = new StreamingContext(sc, Seconds(config.streamingBatchIntervalSeconds))
   val opsPerBatch = (config.numReceivers * config.receiverThroughputPerBatch)
-  val terminationTime = config.totalOps / opsPerBatch * config.streamingBatchIntervalSeconds + 10
+  val estimatedReqRuntime: Long = ((config.totalOps / opsPerBatch) * config.streamingBatchIntervalSeconds) + 10
+  val terminationTime: Long = {
+    if (config.terminationTimeMinutes == 0) {
+      estimatedReqRuntime
+    } else {
+      val newTerminationTime: Long = TimeUnit.MINUTES.toSeconds(config.terminationTimeMinutes)
+      if (estimatedReqRuntime <= newTerminationTime) {
+        println(s"Using the estimated runtime (${estimatedReqRuntime} secs}) required to stream ${config.totalOps} since it is <= the requested runtime (${newTerminationTime} secs).")
+        estimatedReqRuntime
+      } else {
+        println(s"Converting requested runtime of ${config.terminationTimeMinutes} min to ${newTerminationTime} secs.")
+        newTerminationTime
+      }
+    }
+  }
```

Comment:
The spark streaming test already has an interrupt approach to running the workload. It estimates how long we should need to streaming the number of ops specified, then interrupts. If the user specifies a timebox it simply checks against the previously used method for runtime estimation and acts accordingly.

On a side note, at the moment we are not collecting/reporting any metrics for streaming tests. The streaming statistics is not yet available through the REST API but one could enable a metric sink of their choosing and monitor outside this test.